sources/packages/api/src/esdc-integration/fed-restriction-integration/fed-restriction-processing.service.ts

```diff
@@ -0,0 +1,254 @@
+import { InjectLogger } from "../../common";
+import { LoggerService } from "../../logger/logger.service";
+import { Inject, Injectable } from "@nestjs/common";
+import {
+  ConfigService,
+  FederalRestrictionService,
+  RestrictionService,
+} from "../../services";
+import { FedRestrictionIntegrationService } from "./fed-restriction-integration.service";
+import { ESDCIntegrationConfig } from "../../types";
+import * as os from "os";
+import { Connection, InsertResult } from "typeorm";
+import { FederalRestriction, Restriction } from "../../database/entities";
+import { FEDERAL_RESTRICTIONS_BULK_INSERT_AMOUNT } from "../../utilities";
+import { FedRestrictionFileRecord } from "./fed-restriction-files/fed-restriction-file-record";
+import { ProcessSFTPResponseResult } from "./fed-restriction-integration.models";
+
+/**
+ * Manages the process to import the entire snapshot of federal
+ * restrictions, that is received daily, and update the students
+ * restrictions, adding and deactivating accordingly.
+ * * This process does not affect provincial restrictions.
+ */
+@Injectable()
+export class FedRestrictionProcessingService {
+  private readonly esdcConfig: ESDCIntegrationConfig;
+  constructor(
+    @Inject("Connection") private readonly connection: Connection,
+    config: ConfigService,
+    private readonly restrictionService: RestrictionService,
+    private readonly federalRestrictionService: FederalRestrictionService,
+    private readonly integrationService: FedRestrictionIntegrationService,
+  ) {
+    this.esdcConfig = config.getConfig().ESDCIntegration;
+  }
+
+  /**
+   * Import all the federal restrictions from the SFTP and process as below:
+   * 1 - If the restriction is present on federal data and not on the
+   * student data, create a new active restriction;
+   * 2 - If the restriction in present and active on the student data
+   * but it is not present on federal data, deactivate it;
+   * 3 - If the restriction is present on federal data and it is also
+   * present and active on student data, update the updated_at only.
+   */
+  async process(): Promise<ProcessSFTPResponseResult> {
+    //const results: ProcessSftpResponseResult[] = [];
+    // Get the list of all files from SFTP ordered by file name.
+    const fileSearch = new RegExp(
+      `^${this.esdcConfig.environmentCode}CSLS.PBC.RESTR.LIST.D[\w]*\.[0-9]*`,
```

Comment:
are not not including extenstion, `.DAT`

Reply:
Each file follows its own naming conventions. 
![image](https://user-images.githubusercontent.com/61259237/144671141-6126e83f-b6f7-4b7c-b420-eb988b491bc0.png)

Reply:
@abschwenker, can you please double-check if the files are going to come with or without .DAT?
Is there a pattern that we can assume for all files under ESDC?